simplified-books-database/src/main/java/org/nypl/simplified/books/book_database/DatabaseFormatHandleAudioBook.kt

```diff
@@ -240,6 +254,27 @@ internal class DatabaseFormatHandleAudioBook internal constructor(
     this.parameters.onUpdated.invoke(newFormat)
   }
 
+  override fun copyInBook(file: File) {
+    val newFormat = synchronized(this.dataLock) {
+      if (file.isDirectory) {
+        DirectoryUtilities.directoryCopy(file, this.fileBook)
+      } else {
+        FileUtilities.fileCopy(file, this.fileBook)
+      }
+
+      this.formatRef = this.formatRef.copy(
+        file = this.fileBook,
+        manifest = BookFormat.AudioBookManifestReference(
+          manifestURI = URI("manifest.json"),
+          manifestFile = this.fileManifest
+        )
+      )
+      this.formatRef
```

Comment:
You can remove the assignment and the return value in the end and just leave the

```
this.formatRef.copy(
        file = this.fileBook,
        manifest = BookFormat.AudioBookManifestReference(
          manifestURI = URI("manifest.json"),
          manifestFile = this.fileManifest
        )
      )
```